### PR TITLE
Cloud functions base structure commit

### PR DESCRIPTION
**Pull request by musabmjafri10p**

### CODE REVIEW

**Comment by mohuk** [comment]
The entire code base is in Typescript, wouldn't it make sense to convert cloud functions into Typescript as well ?

**Comment by musabmjafri** [comment]
Makes sense indeed, wasn't aware we were using typescript. Updated the base structure likewise

File: functions/functions/package.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "functions",
+  "scripts": {
+    "lint": "tslint --project tsconfig.json",
+    "build": "tsc",
+    "serve": "npm run build && firebase serve --only functions",
+    "shell": "npm run build && firebase functions:shell",
+    "start": "npm run shell",
+    "deploy": "firebase deploy --only functions",
```
Comment (mohuk) [inline functions/functions/package.json]: there should be a pre-deploy task to transpile all TS to JS

Comment (musabmjafri) [reply functions/functions/package.json]: It doesn't need the pre-deploy segment in package.json, during the init process it adds that with a firebase.json file... it even deploys and runs sexily on the cloud :p

Comment (mohuk) [reply functions/functions/package.json]: Got it

**Pull request merged**